pkg/listeners/listeners_unix.go

```diff
@@ -36,7 +36,7 @@ func Init(proto, addr, socketGroup string, tlsConfig *tls.Config) ([]net.Listene
 		if err != nil {
 			return nil, fmt.Errorf("can't create unix socket %s: %v", addr, err)
 		}
-		ls = append(ls, l)
+		ls = append(ls, &MalformedHostHeaderOverride{l})
```

Comment:
You also need to do this in the case "fd": case above it seems, I added

		for i := range fds {
			fds[i] = &MalformedHostHeaderOverride{fds[i]}
		}